apis/core.oam.dev/v1alpha1/external_types.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+)
+
+// +kubebuilder:object:root=true
+
+// Policy is the Schema for the policy API
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories={oam}
+// +kubebuilder:printcolumn:name="TYPE",type=string,JSONPath=`.type`
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Policy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Type string `json:"type"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Properties *runtime.RawExtension `json:"properties,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// PolicyList contains a list of Policy
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Policy `json:"items"`
+}
+
+// +kubebuilder:object:root=true
+
+// Workflow is the Schema for the policy API
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories={oam}
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Workflow struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Steps []v1beta1.WorkflowStep `json:"steps,omitempty"`
```

Comment:
cross version type imports should be prohibitated

Reply:
Yes. It is currently hard to let the new CRD use the name Policy & Workflow but still keep compatibility for the previous code. I agree we should not do this.